### ISSUE TITLE
WPS networks not displaying as WPS networks 

### ISSUE THREAD

**Issue by NovaCygni** [derv82/wifite2#8]
Im looking into the issue atm of the WPS networks not being detected as WPS thus, unable to be attacked when they should be able to, roughly know whats causing it so ill leave this bug-report open till ive submitted the code changes required to fix this issue (\* will likely change the WPS method in 2 ways, 1) use Wash to confirm Lock status 2) allow a "IgnoreWPSNotDetected" toggle to attempt attacks on routers regardless of WPS status *)

**Comment by weedy** [comment]
This is still broken in master, which basically kills a lot of the reason you would use this.

Please reopen this @NovaCygni

**Comment by yuri-sevatz** [comment]
Possibly related (Pentoo, which uses wifite2 and reaver-wps-fork-t6x)

https://github.com/pentoo/pentoo-overlay/issues/139

... Looks like reaver-wps-fork-t6x inverts the meaning of '-C' command in wash, hence nothing shows up as WPS-capable in wifite2 when using that version.

**Comment by yuri-sevatz** [comment]
@NovaCygni What version of reaver is Wifite2 meant to use?

**Comment by blshkv** [comment]
It looks like you should drop "-C" parameter. The latest version of reaver-wps-fork-t6x ignores it by default (see https://github.com/t6x/reaver-wps-fork-t6x/commit/63c5c02bfc44e732399118d844906b0708e8e28c), and -C flag does exact opposite.

**Comment by blshkv** [comment]
Removing my previous comment.  reaver fork guys seems agreed to revert it:
https://github.com/t6x/reaver-wps-fork-t6x/issues/106

**Comment by NovaCygni** [comment]
@yuri-sevatz It was supposed to use the t6x-github fork but derv appears to have gone awol on this project. If you want im planning on actually redoing the whole Wifite and the required Reaver as a StandAlone "Wifite3" in Python 3.5, so far willing contributors appears to be just myself but id love a extra hand working on it. 
Wifite2 is currently broken thus: 
Entire WPS code section missing, missing "Wash" status, redundant code methods, not auto down/up/monitor mode on all distros. No Error-Logic checking,  faulty resuming from previous scans method. 
Wifite original itself is flawed also, ie, it relies upon Reaver doing the Assositation with AP's this is known to be highly unstable on almost all Push-Button WPS enabled routers and will often results in false attack results that go up to 99.99% but never actually do anything. It also doesnt if a a previous attack on a AP is detected, offer the choice of resuming the old attack, resuming old attack with modified switches compared to before, or starting a new one.

**Comment by weedy** [comment]
I'm highly interested in a new version that has a focus on error checking. 
I've made some ugly edits to a couple versions of wifite on github, all of which made me wonder how it ever worked.

Is py3 out of familiarity?

**Comment by NovaCygni** [comment]
@weedy Partially but also because itl allow Async to be used so when a error is thrown rather than Except to deal with it I can use "Await" to chain the fix for error thrown then continue on which would be alot more efficient as it means fewer dropped exchanges due to timeout... nothing worse than a couple M2 or M3 out of Sync that causes an entire false "Run Through". 

Id love to check the "Ugly Edits" :p Always amazed me what little gems of genius people casually put in to solve x problem :)

**Comment by blshkv** [comment]
Please do not call wash with "-C" parameter. This is a default behaviour now and the flag has been removed

**Comment by derv82** [comment]
Also got this error. Removed the `-C` option in the commit above. Should resolve the issue.

**Comment by wifiuk** [comment]
i was getting WPS networks displayed until the recent change, now i get none

**Comment by derv82** [comment]
@wifiuk 

0. What OS are you running? What version of `wash` is installed (`wash 2>/dev/null | head -5`)?
1. Are you seeing `n/a` or `no` under the `WPS` column?
2. Do any networks appear with the `--wps` option?
3. Can you run with `-vvv` (very verbose mode) and paste the output of the `wash` command? Full output might be useful as well.

**Comment by wifiuk** [comment]
wps is displaying fine with the new change

the change before it was displaying no under the wps column


**wash 2>/dev/null | head -5**
Wash v1.5.3 WiFi Protected Setup Scan Tool
Copyright (c) 2011, Tactical Network Solutions, Craig Heffner
mod by t6_x<t6_x@hotmail.com>, DataHead, Soxrok2212, Wiire, AAnarchYY & rofl0r

**Comment by derv82** [comment]
Ok cool. Closing this issue, but anyone can reopen if it's still a problem.

**Comment by blshkv** [comment]
@derv82 users are getting lost between wifite(which is outdated and broken), wifite-fork (with non-technical and unfriendly developers) and wifite2. Can take an initiative, port back any changes from the fork and release a one version which actually works?

**Comment by derv82** [comment]
@blshkv Are you asking for an update to the old wifite?

I'm choosing to ignore the old version of `wifite` indefinitely. It was one of my first Python scripts I wrote while learning programming, and it's 3.5K lines of ugly, untested, unmaintainable code. 

I created #19 for tracking the backport of features from wifite forks into this repo (`wifite2`). Once this repo is "good enough" and has feature-parity, I will update [the old Wifite repo](https://github.com/derv82/wifite) to point users here.

Feel free to create a new issue for each feature you want Wifite2 to have.
